src/rust/engine/client/src/build_root.rs

```diff
@@ -0,0 +1,165 @@
+// Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+// Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+use log::debug;
+use std::env;
+use std::ops::Deref;
+use std::path::PathBuf;
+
+#[derive(Debug)]
+pub struct BuildRoot(PathBuf);
+
+impl BuildRoot {
+  const SENTINEL_FILES: [&'static str; 3] = ["pants", "BUILDROOT", "BUILD_ROOT"];
```

Comment:
Nope, that won't compile.

Reply:
The following works in the Rust Playground:

```
const STRINGS: &[&'static str] = &["foo", "bar", "ha"];

fn main() {
    println!("{:?}", STRINGS);
}
```

So `&[&'static str]` instead of `[&'static str]`.

Reply:
This works:
```rust
const SENTINEL_FILES: &'static [&'static str] = &["pants", "BUILDROOT", "BUILD_ROOT"];
```
I'm not really sure why it's better though.

Reply:
My thought was because it removes having to explicitly state the length of the slice.